content/docs/atom-guide/look-dev/vegetation-bending-tutorial.md

```diff
@@ -0,0 +1,716 @@
+
+### Render the material at an offset
+To start off, let's edit the vertex shader to render our shader ball at an offset.
+1. Open `{your-project-path}\Materials\Types\VegetationBending_ForwardPass.azsli`. Recall that `.azsli` files contain shader code. This file contains the vertex and pixel shader code for the *forward pass* of the vegetation bending material type.
+2. Find the function `VegetationBending_ForwardPassVS`.
+3. Towards the end of the function, right before `OUT.m_worldPosition = worldPosition.xyz;`, add: 
+   
+   ```hlsl
+   worldPosition.x += 5;
+   ```
+   This will adjust the position in the positive x direction by 5 units. You may wonder why we are editing `worldPosition` instead of `m_position`; `m_position` is the position of this vertex relative to the origin of the model, whereas `m_worldPosition` is the position of this vertex relative to the origin of the level (or world). Try out editing the other dimensions and `m_position` and `worldPosition` and see what they do!
+4. Make sure the **Editor** is open, if it is not already open.
+5. Save your file with **CTRL-S** and the **Asset Processor** should automatically detect changes and process the file. You can open the **Asset Processor** and check when the file is done processing. 
+   {{< note >}}
+   If you can't find the **Asset Processor**, click the arrow on the bottom right of your taskbar, and then click on the icon of two arrows in a circle.
+   {{< /note >}}
+6. Once the **Asset Processor** is done processing the changes, you should see in the **Editor** that your material looks different!
+
+{{< image-width src="/images/atom-guide/vegetation-bending-tutorial/offset.png" width="100%" alt="Forward pass offset." >}}
+
+The main texture of our shader ball seems to be showing up at an offset as we intended, but a grey outline is still at the origin of the object. This is because we only edited the forward pass, but have yet to edit the *depth pass*. All the passes we go through are referenced in `VegetationBending.materialtype`.
+More information about passes can be found [here](../dev-guide/passes/), but keep in mind that different passes render different parts of the material, and some passes' outputs are used as inputs to other passes.
+
+Let's repeat the above steps but with the depth pass:
+1. Open `{your-project-path}\Materials\Types\VegetationBending_DepthPass.azsli`.
+   - Make sure you are not editing the `VegetationBending_DepthPass_WithPS.azsli` file.
+2. Find the function `DepthPassVS`.
+3. Towards the end of the function, right before `OUT.m_position = mul(ViewSrg::m_viewProjectionMatrix, worldPosition);`, add: 
+   
+   ```
+   worldPosition.x += 5;
+   ```
+4. Save your file and check our the **Editor**. The shader ball should now be completely rendered at an offset! 
+   - Note that the shadow is still in the original position. After finishing this tutorial, see if you can adjust the shadow yourself!
```

Comment:
I think that this is a bit unclear, since you will fix the shadow when you add the shadowmap_with_ps. Though if I follow correctly, we are not covering the non-ps version of the shadowmap shader in the tutorial. I think something along these lines should point them more in the right direction: 

"Note that the shadow is still in the original position. We'll be correcting this later on in the tutorial when we add a custom shadowmap with pixel shader <link to the step in the tutorial where this happens>. The final versions of the tutorial files in the <sample-code-repo link> include a vertex-only shadowmap shader, but for brevity, we are not covering it in the tutorial. After finishing this tutorial, see if you can adjust the  vertex-only shadow (used for opaque materials that don't have alpha-cutout) yourself before comparing your work to the final versions!"

Reply:
This makes sense and is a good suggestion, but I didn't include a vertex-only shadow shader but only used the StandardPbr ones so I changed it to:

"Note that the shadow is still in the original position. We'll be correcting this later on in the tutorial when we add a custom shadowmap with pixel shader. For brevity, we are not covering the vertex-only shadowmap shader in the tutorial and instead using the StandardPBR one. After finishing this tutorial, see if you can add and adjust the vertex-only shadow (used for opaque materials that don't have alpha-cutout) yourself!"

Reply:
👍